src/icq/poller.rs

```diff
@@ -180,26 +182,47 @@ pub async fn process_event_hist_dlg_state(
             .clone()
     };
 
+    process_hist_dlg_state_messages(
+        tx,
+        session,
+        account_info.handle.clone(),
+        &event_data.sn,
+        &event_data.persons,
+        &event_data.mchat_state,
+        &event_data.messages,
+    )
+    .await;
+}
+
+pub async fn process_hist_dlg_state_messages(
+    mut tx: FdSender<SystemMessage>,
+    session: SessionInfo,
+    handle: Handle,
+    event_sn: &str,
+    event_persons: &[events::HistDlgStatePerson],
+    event_mchat_state: &Option<events::HistDlgStateMChatState>,
```

Comment:
I think it is better to use `Option<&event::HistDlgStateMChat>`. 

Rust is able to optimize `Option<&_>` as a single pointer where `None` == `null`. With `&Option<_>` this compile to a pointer to a pointer.

Reply:
(Some fun compiler explorer: https://godbolt.org/z/edYe88 , note the difference between f2 before the unwrap and in caller)

Reply:
And then do I have to create a new conversion method?

```
the trait bound `chat_info::ChatInfoVersion: std::convert::From<&icq::client::events::HistDlgStateMChatState>` is not satisfied
```